### PR TITLE
Make `ILayoutRestorer` optional in the extension tutorial

### PR DESCRIPTION
**Pull request by jtpio**
## References

Fixes https://github.com/jupyterlab/jupyterlab/issues/11611.

The motivation here is to make extension authors more aware of the `optional` field of the plugin definition, so they can build extensions that can be used in more JupyterLab based applications by default.

## Code changes

Documentation changes.

- Make `ILayoutRestorer` an `optional` to cover `optional` in the extension tutorial
- Keep `ICommandPaletter` as `requires` to also cover tokens specified as `requires`.

## User-facing changes

None for end users.

Documentation changes can be previewed on RTD: https://jupyterlab--11677.org.readthedocs.build/en/11677/extension/extension_tutorial.html

## Backwards-incompatible changes

None

### CODE REVIEW

**Comment by jtpio** [comment]
A follow-up to this would be to update the apod repo to take these changes into account: https://github.com/jupyterlab/jupyterlab_apod

File: docs/source/extension/extension_tutorial.rst [hunk 1]
```diff
@@ -792,30 +792,43 @@ entire list of import statements looks like the following:
...
+
+    You can learn more about ``requires`` and ``optional`` in the :ref:`tokens` section
+    of the Extension Developer Guide.
+
 Finally, rewrite the ``activate`` function so that it:
 
 1. Declares a widget variable, but does not create an instance
    immediately.
-2. Constructs a ``WidgetTracker`` and tells the ``ILayoutRestorer``
+2. Adds the global ``LayoutRestorer`` as the thid parameter of the ``activate`` function.
```
Comment (fcollonval) [inline docs/source/extension/extension_tutorial.rst:823]:
```suggestion
2. Adds the global ``LayoutRestorer`` as the third parameter of the ``activate`` function.
```

**Review by fcollonval — COMMENTED** [review]
Thanks @jtpio 

I caught a typo. Otherwise LGTM

File: docs/source/extension/extension_tutorial.rst (hunk 1 above)
Comment (jtpio) [reply docs/source/extension/extension_tutorial.rst:823]: Good catch, thanks!

**Review by fcollonval — APPROVED** [review]
LGTM

**Comment by fcollonval** [comment]
I think we should wait to solve jupyterlab/jupyterlab_apod#10 before backporting it to 3.2.x

**Pull request merged**